net/json_handler.go

```
@@ -0,0 +1,183 @@
+
+	h.WebHook.channel <- &m
+}
+
+//ConfigFromFile reads the config.yaml
+func ConfigFromFile(filename string) (cfg *WebHookConfig, err error) {
+	logr.Infof("Loading configuration at '%s'", filename)
+	configFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("can't open the config file: %s", err)
+	}
+	config := WebHookConfig{}
+
+	err = yaml.Unmarshal(configFile, &config)
+	if err != nil {
+		return nil, fmt.Errorf("can't read the config file: %s", err)
+	}
+	logr.Info(config)
+	logr.Info("Configuration loaded")
+	return &config, nil
+}
+
+func (h *JSONHandler) processAlerts() {
+
+	logr.Info("Alerts queue started")
+
+	// While there are alerts in the queue, batch them and send them over to Zabbix
+	var metrics []*zabbix.Metric
+	for {
+		select {
+		case r := <-h.WebHook.channel:
+			if r == nil {
+				logr.Info("Queue Closed")
+				return
+			}
+
+			host := h.WebHook.config.ZabbixHostAnnotation
+			// Send alerts only if a host annotation is present
+			if host != "" {
+				key := fmt.Sprintf("%s.%s", h.WebHook.config.ZabbixKeyPrefix, strings.ToLower(r.CommonLabels["alertname"]))
+				value := "0"
+				if r.Status == "firing" {
+					value = "1"
+				}
+
+				logr.Infof("added Zabbix metrics, host: '%s' key: '%s', value: '%s'", host, key, value)
+				metrics = append(metrics, zabbix.NewMetric(host, key, value))
+			}
+		default:
+			if len(metrics) != 0 {
+				h.zabbixSend(metrics)
+				metrics = metrics[:0]
+			} else {
+				time.Sleep(1 * time.Second)
+			}
+		}
+	}
+}
+func (h *JSONHandler) zabbixSend(metrics []*zabbix.Metric) {
+	// Create instance of Packet class
+	packet := zabbix.NewPacket(metrics)
+
+	// Send packet to zabbix
+	logr.Infof("sending to zabbix '%s:%d'", h.WebHook.config.ZabbixServerHost, h.WebHook.config.ZabbixServerPort)
+	z := zabbix.NewSender(h.WebHook.config.ZabbixServerHost, h.WebHook.config.ZabbixServerPort)
+	res, err := z.Send(packet)
+	fmt.Println(string(res))
+	if err != nil {
+		logr.Error(err)
```

return an error, rather than logging it here